### ISSUE TITLE
RuntimeError: cannot join current thread

### ISSUE THREAD

**Issue by garbb** [danpaquin/GDAX-Python#51]
Running python 2.7.13 on windows 7 x64.

Running the following code gets a random crash after anywhere between 12-16 hours later.
```python
import GDAX

class myWebsocketClient(GDAX.WebsocketClient):
	def onOpen(self):
		self.url = "wss://ws-feed.gdax.com/"
		self.products = ["BTC-USD"]
	def onMessage(self, msg):
		if 'price' in msg and 'type' in msg and msg["type"] == 'match':
			print msg['time'], float(msg['price'])

			
wsClient = myWebsocketClient()
wsClient.start()
```

Crash
```
-- Socket Closed --
Exception in thread Thread-1:
Traceback (most recent call last):
  File "C:\Python27\lib\threading.py", line 801, in __bootstrap_inner
    self.run()
  File "C:\Python27\lib\threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "build\bdist.win32\egg\GDAX\WebsocketClient.py", line 28, in _go
    self._listen()
  File "build\bdist.win32\egg\GDAX\WebsocketClient.py", line 57, in _listen
    self.close()
  File "build\bdist.win32\egg\GDAX\WebsocketClient.py", line 68, in close
    self.thread.join()
  File "C:\Python27\lib\threading.py", line 931, in join
    raise RuntimeError("cannot join current thread")
RuntimeError: cannot join current thread
```

**Comment by garbb** [comment]
Well I at least have found an easy way to re-create this crash. Run the code I posted and then close the connection from python.exe with [tcpview](https://technet.microsoft.com/en-us/sysinternals/tcpview.aspx) (this is for windows) and the crash occurs.

**Comment by garbb** [comment]
This exception can be avoided by removing "self.thread.join()" from line 68 of GDAX-0.3.1-py2.7.egg\GDAX\WebsocketClient.py.
Then when the connection drops, the .onClose() that you write when you create your websocketclient will be called so you can handle re-connecting or whetever you want.

Although I have no idea if this is the best solution or if it will cause other problems...

**Comment by danpaquin** [comment]
This line has been removed.  We should consider options to add this back without the possibility of errors.

**Comment by royitaqi** [comment]
Hi @danpaquin ,

How is this problem going? I tried to subscribe to "user" channel (**authenticated**) and the connection closed very soon. Retried many times and they all die after the same amount of time. Looked like it was a heartbeat problem. Below is my log output:

```
$ python gdax_user.py -a int
00:00:00,385  I  Publishing to: tcp://*:5658
00:00:00,385  I  GdaxUser: opening
00:00:00,933  I  GdaxUser: received message: {'channels': [{'name': 'user', 'product_ids': ['ETH-USD']}], 'type': 'subscriptions'}
Connection is already closed. - data: None
00:01:00,473  C  GdaxUser: closing
```

Look at the timestamp, apparently the connection was closed from the server side at 1-minute mark.

I tried a similar piece of code but subscribe to "level2" and/or "ticker" channels and they seem to work fine. The problem appear to only happen when I'm subscribing to "user" channel.
